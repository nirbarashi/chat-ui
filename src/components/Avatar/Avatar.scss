$black: #000;

/*profile */
.profile{
  &-image{display:block; width:50px; height:50px; object-fit:cover; background-color:$black; margin-right:10px; border-radius:50%;}
}