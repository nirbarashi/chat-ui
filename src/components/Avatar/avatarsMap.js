import snorlax from "../../assets/avatars/001-snorlax.png";
import psyduck from "../../assets/avatars/002-psyduck.png";
import pikachu from "../../assets/avatars/003-pikachu.png";
import jigglypuff from "../../assets/avatars/004-jigglypuff.png";
import bullbasaur from "../../assets/avatars/005-bullbasaur.png";

export default { snorlax, psyduck, pikachu, jigglypuff, bullbasaur };