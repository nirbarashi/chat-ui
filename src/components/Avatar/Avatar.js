import React from 'react';
import "./Avatar.scss";
import avatarsMap from "./avatarsMap";

const Avatar = ({ avatar }) => {
	return (
		<img className="profile-image" src={avatarsMap[avatar]} alt="" />
	)
}

export default Avatar;