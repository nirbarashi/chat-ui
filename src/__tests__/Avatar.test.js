import React from "react";
import { create } from "react-test-renderer";
import Avatar from "../components/Avatar/Avatar";


describe("Avatar component", () => {
	test("Matches the snapshot", () => {
		const avatar = create(<Avatar avatar="psyduck" />);
		expect(avatar.toJSON()).toMatchSnapshot();
	});

	test("Matches the snapshot2", () => {
		const avatar = create(<Avatar avatar="snorlax" />);
		expect(avatar.toJSON()).toMatchSnapshot();
	});
});