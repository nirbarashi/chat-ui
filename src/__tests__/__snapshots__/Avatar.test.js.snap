// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Matches the snapshot 1`] = `
<img
  alt=""
  className="profile-image"
  src="002-psyduck.png"
/>
`;

exports[`Avatar component Matches the snapshot2 1`] = `
<img
  alt=""
  className="profile-image"
  src="001-snorlax.png"
/>
`;
